storage/src/merkle_tree_db_adapter.cpp

```diff
@@ -305,22 +202,109 @@ Status DBAdapter::getKeyByReadVersion(BlockId version,
 
 BlockId DBAdapter::getLatestBlock() const {
   // Generate maximal key for type 'BlockId'.
-  const auto key = DBAdapterBase::getLatestBlock(DBKeyManipulator::genBlockDbKey(std::numeric_limits<BlockId>::max()));
-  if (key.empty()) {  // no blocks
-    return 0;
+  const auto maxBlockKey = DBKeyManipulator::genBlockDbKey(std::numeric_limits<BlockId>::max());
+  auto iter = db_->getIteratorGuard();
+  auto foundKey = iter->seekAtLeast(maxBlockKey).first;
+  if (iter->isEnd()) {
+    // Assume the last element in the whole range.
```

Comment:
I have changed it to:

 If there are no keys bigger than or equal to the maximal block key, use the last key in the whole range as it may be the actual latest block key. That will allow us to find the latest block even if there are no other bigger keys in the system. Please see db_types.h for key ordering.